### PR TITLE
AI Voice Assistant 

### PR DESCRIPTION
**Pull request by haitomns4173**
## Describe your project in 1-3 sentences

I am making a voice assistant that will be used to answer queries as well as home automation.
I will use a Raspberry Pi board for this project.

## Hack Club Slack handle

Haitomns G

## Checklist before submitting proposal:

- [x] I am a current high school student (or younger).
- [x] I have submitted an RSVP at <https://hackclub.com/winter>
- [x] I am a member of the Hack Club Slack (Join at <https://hackclub.com/slack>)
- [x] I have self-reviewed my project proposal.

### CODE REVIEW

**Review by bellesea — CHANGES_REQUESTED** [review]
@haitomns4173 thanks so much for sharing your project!! what will you be learning through making this and how long do you think it'll take to do this?

**Comment by haitomns4173** [comment]
I wanted to explore the field of raspberry pi, I have never done projects in it so I was eager to learn about it more, and making that AI assistant will help me learn the basics and know much about it. 

About how long it will take, I will say about a week,

DAY 1: Learn about the components of Raspberry pi.
DAY 2: Set up Raspberry pi and take a tour of Raspberry Pi OS.
DAY 3: Installing the required scripts for Google and Amazon Assistants.
Day 4: Getting the API Keys and Tokens from Amazon and Google Developer's Accounts.
Day 5: Connecting the Raspberry Pi to Smart Home Devices and testing the connectivity.
Day 6: If everything works fine, make an enclosure to house Rasberry Pi in with the speaker.
Day 7: Backup Day if anything goes wrong.

Hope I answered all your question.

**Comment by bellesea** [comment]
> I wanted to explore the field of raspberry pi, I have never done projects in it so I was eager to learn about it more, and making that AI assistant will help me learn the basics and know much about it.
> 
> About how long it will take, I will say about a week,
> 
> DAY 1: Learn about the components of Raspberry pi. DAY 2: Set up Raspberry pi and take a tour of Raspberry Pi OS. DAY 3: Installing the required scripts for Google and Amazon Assistants. Day 4: Getting the API Keys and Tokens from Amazon and Google Developer's Accounts. Day 5: Connecting the Raspberry Pi to Smart Home Devices and testing the connectivity. Day 6: If everything works fine, make an enclosure to house Rasberry Pi in with the speaker. Day 7: Backup Day if anything goes wrong.
> 
> Hope I answered all your questions.

hiiii @haitomns4173 thanks for such a comprehensive response :)) what do you think about adding something "unique" to your AI home voice assistant, like perhaps writing some of your own code or designing your own enclosure? something that makes it such that your project has a little bit more you in it?

**Comment by haitomns4173** [comment]
Yes I already own a desktop computer with all the required tools to setup the Pi so no problem there. I already own a SD Card reader to program the Pi with Raspberry Pi OS. 

Talking about adding things to the assistant, definitely I will make a really good in closure out of Wood or Card board to house all the things into it. It will tell now pretty good. 

Recently I have started to learn about speech synthesis and speech recognisation using python, I have already started building a small Assistant and adding feature to it, Surely gonna test it in the Raspberry Pi and who know someday I will be using my own Assistant to do all my works.

**Comment by bellesea** [comment]
> Yes I already own a desktop computer with all the required tools to setup the Pi so no problem there. I already own a SD Card reader to program the Pi with Raspberry Pi OS.
> 
> Talking about adding things to the assistant, definitely I will make a really good in closure out of Wood or Card board to house all the things into it. It will tell now pretty good.
> 
> Recently I have started to learn about speech synthesis and speech recognisation using python, I have already started building a small Assistant and adding feature to it, Surely gonna test it in the Raspberry Pi and who know someday I will be using my own Assistant to do all my works.

@haitomns4173 thanks for the rapid response! could you please this additional context in the markdown file?

**Comment by haitomns4173** [comment]
Done making changes to the file. Have a look.

File: haitomns4173.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+## Plan
+
+First of all I will set up the OS in the SD card of the Raspberry Pi then power it on.
+The plan is to use a Raspberry Pi as a computation device to handle the assistant SDK from Google or Amazon. 
+I will require a speaker and connect it to the Raspberry Pi and also connect a USB mic to it for taking the input. 
+After setting up the hardware I will connect the pi through the SSH clint and set up the SKD as test if the speaker and mic are working. 
+If everything is attached and working fine we can give a command to the Assistant and it will answer the queries.
+After that I am going to test my custom assistant, with some simple question. 
+Finally if everything is tested and working properly, u will amke a wooden I'd cardboard enclosure to put all the contents in it and make it a single unit.
```
Comment (bellesea) [inline haitomns4173.md:23]:
```suggestion
Finally if everything is tested and working properly, I will make a wooden cardboard enclosure to put all the contents in it and make it a single unit.
```

**Comment by bellesea** [comment]
@haitomns4173 could you please factor in shipping and taxes in your budget? thank you sm!

File: haitomns4173.md (hunk 1 above)
Comment (haitomns4173) [reply haitomns4173.md]: Thank you for correcting that!

**Comment by haitomns4173** [comment]
Hey, I have added the shipping and the tax cost.

**Review by bellesea — APPROVED** [review]
@haitomns4173 thank you for your patience with the review and for pushing the changes!

an awesome member of @hackclub/bank-ops will be reaching out soon with next steps! thank you :)

(btw, we're currently on a reduced skeleton team for the holidays so please give us some time)

**Review by MelSmith104 — COMMENTED** [review]
hey @haitomns4173 ! i love this project idea <3 it sounds like it'll be so much fun to build! i'm reaching out to you on slack right now so that we can chat about next steps!

**Review by MelSmith104 — APPROVED** [review]

**Pull request merged**